modules/install/pages/thp-disable.adoc

```diff
@@ -50,83 +36,202 @@ The init script is designed to make sure that the changes are made around the sa
-  elif [ -d /sys/kernel/mm/redhat_transparent_hugepage ]; then
-    echo 'never' > /sys/kernel/mm/redhat_transparent_hugepage/enabled
-    echo 'never' > /sys/kernel/mm/redhat_transparent_hugepage/defrag
-  else
-    return 0
-  fi
-;;
+  start)
+    if [ -d /sys/kernel/mm/transparent_hugepage ]; then
+      thp_path=/sys/kernel/mm/transparent_hugepage
+    elif [ -d /sys/kernel/mm/redhat_transparent_hugepage ]; then
+      thp_path=/sys/kernel/mm/redhat_transparent_hugepage
+    else
+      return 0
+    fi
+
+    echo 'never' > ${thp_path}/enabled
+    echo 'never' > ${thp_path}/defrag
+
+    re='^[0-1]+$'
+    if [[ $(cat ${thp_path}/khugepaged/defrag) =~ $re ]]
+    then
+      # RHEL 7
+      echo 0  > ${thp_path}/khugepaged/defrag
+    else
+      # RHEL 6
+      echo 'no' > ${thp_path}/khugepaged/defrag
+    fi
+
+    unset re
+    unset thp_path
+    ;;
 esac
 ----
++
+Save and close your editor.
 
-Register the code in the OS:: As root do the following:
-. Create a file with the above code:
+. Make the script executable.
 +
+[source,console]
 ----
-# vi /etc/init.d/disable-thp
+sudo chmod 755 /etc/init.d/disable-thp
 ----
 
-. Chmod the file to be executable:
+. Configure the OS to run the script on boot.
++
+[{tabs}] 
+==== 
+Red Hat, CentOS, & Amazon Linux::
 +
+--
+[source,console]
 ----
-# chmod 755 /etc/init.d/disable-thp
+sudo chkconfig --add disable-thp
 ----
+--
 
-. Execute it so that it takes effect right now:
-+
+Ubuntu & Debian:: 
++ 
+--
+[source,console]
 ----
-# service disable-thp start
+sudo update-rc.d disable-thp defaults
 ----
+--
+
+SUSE::
 +
-[NOTE]
+--
+[source,console]
+----
+sudo insserv /etc/init.d/disable-thp
+----
+--
 ====
-If you encounter an error running the script, you can run the commands manually as follows:
 
+. Override `tuned` and `ktune`, if necessary.
++
+If you are using `tuned` or `ktune` (for example, if you are running Red Hat/CentOS 6+) you must also <<tuned-ktune,configure them to preserve the above settings after reboot>>.
+
+. Reboot the system and <<verify-thp,verify that THP is disabled>>.
+
+[#tuned-ktune]
+=== If Using `tuned` and `ktune`
+
+`tuned` and `ktune` are system monitoring and tuning tools available on Red Hat and CentOS.
+When they are in use on a system, they can be used to enable and disable THP.
+
+To disable THP in `tuned` and `ktune`, you need to edit or create a new _profile_ that sets THP to `never`.
+
+[{tabs}] 
+==== 
+Red Hat/CentOS 7:: 
++
+--
+. Create a new `tuned` directory for the new profile.
++
+[source,console]
 ----
-# echo never > /sys/kernel/mm/transparent_hugepage/enabled
-# echo never > /sys/kernel/mm/transparent_hugepage/defrag{{}}
+sudo mkdir /etc/tuned/no-thp
 ----
-====
 
-Make sure the Init script starts at boot::
-For Red Hat variants:
+. Create and edit `tuned.conf`.
 +
+[source,console]
 ----
-sudo chkconfig disable-thp on
+vi /etc/tuned/no-thp/tuned.conf
 ----
 +
-For Ubuntu:
+Add the following contents:
 +
+[source,console]
```

Comment:
Perhaps create low priority docs-ui issue to add generic conf files to `highlight.js`.